ndmg/stats/qa_tensor.py

```diff
@@ -64,15 +67,15 @@ def generate_3_d_directions(peak_dirs,peak_values):
     heights = []
     directions_colors = []
     
-    #iteratively create required new arrow entries per value in 3-d peak object
-    for x in range(peak_dirs.shape[0]):
-        for y in range(peak_dirs.shape[1]): 
-            for z in range(peak_dirs.shape[2]):
-                centers.append([x,y,z])
+    xs = range(peak_dirs.shape[0])
+    ys = range(peak_dirs.shape[1])
+    zs = range(peak_dirs.shape[2])
+    for x, y, z in itertools.product(xs, ys, zs):
+        centers.append([x,y,z])
                 
-                #relies on the fact that peaks_from_models generates peak_dirs and peak_values in descending order
-                directions.append(peak_dirs[x,y,z,0,:])
-                heights.append(peak_values[x,y,z,0])
+        #relies on the fact that peaks_from_models generates peak_dirs and peak_values in descending order
+        directions.append(peak_dirs[x,y,z,0,:])
+        heights.append(peak_values[x,y,z,0])
```

Comment:
cool, looks better (generally, [flat is better than nested](https://www.python.org/dev/peps/pep-0020/)). 

I think devin's numpy code would be a bit more computationally efficient, but probably it's not a big enough deal to make a fuss about. 

I'm waiting on the pipeline to run right now, so I'll comment again once I get to this so that I can run through it in the debugger.